blocks/components/editable/index.js

```diff
@@ -349,6 +350,8 @@ export default class Editable extends wp.element.Component {
 		this.setState( {
 			formats: merge( {}, this.state.formats, formats )
 		} );
+
+		this.editor.setDirty( true );
```

Comment:
I decided to remove the setDirty call since it seems controversial. Note that applying formatting is very buggy/broken on Firefox but I don't know how to address that and it is not really anything to do with the list block.